### PR TITLE
feat: allow to config cors async

### LINKED ISSUE #6118: Configuring CORS Asynchronously
## Is your feature request related to a problem? Please describe.
The "cors" library allows us to set the [configuration asynchronously](https://github.com/expressjs/cors#configuring-cors-asynchronously). This cannot be use with enableCors method, we need to do some workarounds.

## Describe the solution you'd like
It would be great to have enableCorsAsync() or if enableCors method could receive also this option property
```typescirpt
asyncConfiguration: (req) => Promise<CorsOptions>
```

## Teachability, Documentation, Adoption, Migration Strategy
It would be great to have enableCorsAsync() or if enableCors method could receive also this option property
```typescirpt
app.enableCors({asyncConfiguration: (req) => {
  return {origin: !isInternalRequest(req)};
})
```

It should require any migration.

## What is the motivation / use case for changing the behavior?
I'm using app.enableCors() to add cors headers to responses from my application, but in some cases, I don't want them, f. ex. when the request comes from the internal network.

### PR DESCRIPTION
**Pull request by Sikora00**
Bump version of the fastify-cors.
Add CorsOptionsDelegate interface.
Add CorsOptionsDelegate as an acceptable type
for every enableCors usage.

Closes #6118

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #6118

## Does this PR introduce a breaking change?
```
[] Yes
[X] No
```

## Other information

### CODE REVIEW

File: integration/hello-world/e2e/cors.spec.ts [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+            });
+        });
+
+        after(async () => {
+          await app.close();
+        });
+      });
+    });
+  });
+  describe('Express', () => {
```
Comment (jmcdo29) [inline integration/hello-world/e2e/cors.spec.ts:237]: I would suggest separating the Fastify and Express tests. Keeps the test files smaller. If you re-use any main functions between the two (like the async config) you can put it into a test util file as well and make it more easily re-usable

**Review by jmcdo29 — APPROVED** [review]

**Comment by jmcdo29** [comment]
Cool, everything looks pretty good to me. I'll let Kamil take a look and give his two cents. Looks like a nice update.

**Comment by kamilmysliwiec** [comment]
LGTM thanks! I'll kick off a new release with this PR very shortly

**Comment by rubiin** [comment]
cant wait for the next release, good things are on the way :)

**Pull request merged**